provider/balance_checker_test.go

```diff
@@ -2,44 +2,59 @@ package provider
 
 import (
 	"context"
-	"github.com/boz/go-lifecycle"
+	"testing"
+	"time"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	bankTypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	tmrpc "github.com/tendermint/tendermint/rpc/core/types"
+
 	"github.com/ovrclk/akash/provider/event"
 	"github.com/ovrclk/akash/provider/session"
 	"github.com/ovrclk/akash/pubsub"
 	"github.com/ovrclk/akash/testutil"
+	dtypes "github.com/ovrclk/akash/x/deployment/types/v1beta2"
+	"github.com/ovrclk/akash/x/escrow/types/v1beta2"
+	mtypes "github.com/ovrclk/akash/x/market/types/v1beta2"
 	ptypes "github.com/ovrclk/akash/x/provider/types/v1beta2"
-	"github.com/stretchr/testify/mock"
-	"github.com/stretchr/testify/require"
-	"testing"
-	"time"
 
+	akashmock "github.com/ovrclk/akash/client/mocks"
 	cosmosMock "github.com/ovrclk/akash/testutil/cosmos_mock"
 )
 
+type broadcaster struct {
+	txCh chan sdk.Msg
+}
+
 type scaffold struct {
-	testAddr    sdk.AccAddress
-	testBus     pubsub.Bus
-	ctx         context.Context
-	cancel      context.CancelFunc
-	queryClient *cosmosMock.QueryClient
-	bc          *balanceChecker
+	testAddr  sdk.AccAddress
+	testBus   pubsub.Bus
+	ctx       context.Context
+	cancel    context.CancelFunc
+	qc        *cosmosMock.QueryClient
+	aqc       *akashmock.QueryClient
+	bc        *balanceChecker
+	broadcast *broadcaster
 }
 
-func (s *scaffold) start() {
-	go s.bc.lc.WatchContext(s.ctx)
-	go s.bc.run()
+func (bc *broadcaster) Broadcast(_ context.Context, msg ...sdk.Msg) error {
+	bc.txCh <- msg[0]
```

Comment:
why is it variadic?

Reply:
https://github.com/ovrclk/akash/blob/e6e207f79ea812407f99ab89f3b75c299ee1f7b5/client/broadcaster/client.go#L22